cirq/study/sweepable.py

```diff
@@ -55,7 +55,24 @@ def to_sweeps(sweepable: Sweepable) -> List[Sweep]:
     if isinstance(sweepable, Sweep):
         return [sweepable]
     if isinstance(sweepable, dict):
-        return [_resolver_to_sweep(ParamResolver(cast(Dict, sweepable)))]
+        expandsweepable = [cast(dict, {})]
+        for key in sweepable.keys():
+            value = sweepable[key]
+            if isinstance(value, Iterable):
+                tempsweepable = []
+                for item in value:
+                    for shortdict in expandsweepable:
+                        tempdict = shortdict.copy()
+                        tempdict[key] = item
+                        tempsweepable.append(cast(dict, tempdict))
```

Comment:
The above code is quite tricky to understand.  It took me a while to figure out what this was doing, since it is repeatedly assigning the values.  I would echo Victory's comment for using itertools if possible.  This will also avoid recreating the entire dictionary every time there is a new list found as well.